integrations/utils/commonUtils.js

```diff
@@ -163,6 +163,40 @@ function getDestinationExternalID(message, type) {
   return destinationExternalId;
 }
 
+function getIntgCommonNames(intg) {
+  const cNamesMap = {};
+  const { NAME, DISPLAY_NAME, CNameMapping } = intg;
+
+  let nameSanitized;
+  if (NAME) {
+    nameSanitized = NAME.trim();
+    cNamesMap[nameSanitized.toLowerCase()] = nameSanitized;
+  }
+
+  if (DISPLAY_NAME) {
+    const displayNameSanitized = DISPLAY_NAME.toLowerCase().trim();
+    // Add the sanitized display name
+    cNamesMap[displayNameSanitized] = nameSanitized;
+
+    const words = displayNameSanitized.split(" ");
+
+    cNamesMap[words.join("_")] = nameSanitized;
+    cNamesMap[words.join("")] = nameSanitized;
+  }
+
+  // add the hard-coded common names
+  if (CNameMapping) {
+    // convert the keys to lower case to
+    // match the existing common names
+    Object.keys(CNameMapping).forEach((name) => {
+      const sanitizedName = name.toLowerCase().trim();
+      cNamesMap[sanitizedName] = CNameMapping[name];
```

Comment:
`cNamesMap[sanitizedName] = CNameMapping[name];` this will create duplicate records as we are not removing the existing key-value. Is this expected?

Reply:
It'll override if a key is already present.